extras/i2c/i2c.h

```diff
@@ -27,16 +27,39 @@
 
 #include <stdint.h>
 #include <stdbool.h>
+#include <errno.h>
+#include <FreeRTOS.h>
+#include <task.h>
 
 #ifdef	__cplusplus
 extern "C" {
 #endif
 
+
+#define I2C_FREQUENCY_400K true  // for test WIP
+
+/*
+ *  Some bit can be transmit slower.
+ *  Selected frequency fix the speed of a bit transmission
+ *  I2C lib take the maximum frequency defined
+ */
+
+#ifdef I2C_FREQUENCY_500K
+#define I2C_CUSTOM_DELAY_160MHZ   6
+#define I2C_CUSTOM_DELAY_80MHZ    1   //Sry, maximum is 320kHz at 80MHz
```

Comment:
It is like a good idea to get performance like this  and store frequency at `i2c_start() `for example but the link will be less secure for a special case (Probability very low, but murphy is here ). 
When you start a transaction, if the task change and second task change frequency ( 80 => 160 ) , when the i2C link come back, delay will be very low and frequency will be high ( approximately  800KHz for 400KHz base link ). Can expect EIO error.  Of course user can prevent this manually with status from i2c bus when he need change frequency or doing some treatments from this error. The component can support high frequency too.